src/Python/qsharp/tests/test_azure.py

```diff
@@ -51,19 +51,31 @@ def test_empty_workspace():
     jobs = qsharp.azure.jobs()
     assert jobs == []
 
-def test_workspace_with_providers():
+def test_create_workspace_with_parameters():
     """
-    Tests behavior of a mock workspace with mock providers.
+    Tests creation of a mock workspace with mock providers using parameters.
```

Comment:
You could also consider factoring out the part of `test_workspace_with_providers` that tests job submission, so that you could have a full test for both a workspace created with separate parameters and one created with a resource ID.

Reply:
Good idea. And actually by doing that I just found a bug related to connecting to multiple workspaces in one session. 😊